pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -118,12 +116,12 @@ func (o *EditOptions) Complete(f cmdutil.Factory, args []string, cmd *cobra.Comm
 	if o.EditMode != NormalEditMode && o.EditMode != EditBeforeCreateMode && o.EditMode != ApplyEditMode {
 		return fmt.Errorf("unsupported edit mode %q", o.EditMode)
 	}
-	if o.Output != "" {
-		if o.Output != "yaml" && o.Output != "json" {
-			return fmt.Errorf("invalid output format %s, only yaml|json supported", o.Output)
+	if *o.PrintFlags.OutputFormat != "" {
+		if *o.PrintFlags.OutputFormat != "yaml" && *o.PrintFlags.OutputFormat != "json" {
```

Comment:
I'd like to use the error handling and printer matching capabilities of `PrintFlags` where possible. Rather than explicitly returning a JSON / YAML printer directly in this command, I had something [like this](https://github.com/charrywanganthony/kubernetes/pull/2) in mind.

cc @deads2k

Reply:
> juanvallejo 18 minutes ago  Member
> I'd like to use the error handling and printer matching capabilities of PrintFlags where possible. Rather than explicitly returning a JSON / YAML printer directly in this command, I had something like this in mind.

yeah, your pull (2) is closer to what I'd expect.